subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/fingerprint/ConfigurationCacheFingerprintChecker.kt

```diff
@@ -202,6 +203,9 @@ class ConfigurationCacheFingerprintChecker(private val host: Host) {
                     return "remote script $uri has changed"
                 }
             }
+            is ConfigurationCacheFingerprint.FileSystemDefaultExcludes -> input.run {
+                host.updateFileSystemDefaultExcludes(excludes.toTypedArray())
```

Comment:
DefaultExcludes can be changed in settings file. So if settings file was changed - configCache will got invalidated anyway. So I've considered it's barely possible to fail with checking was defaultExclude changed or not, and decided to update it at each check.